src/Magnum/Shaders/Test/FlatGLTest.cpp

```diff
@@ -853,6 +860,37 @@ template<UnsignedInt dimensions> void FlatGLTest::construct() {
     MAGNUM_VERIFY_NO_GL_ERROR();
 }
 
+template<UnsignedInt dimensions> void FlatGLTest::constructAsync() {
+    setTestCaseTemplateName(Utility::format("{}", dimensions));
+
+    auto&& data = ConstructData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    #ifndef MAGNUM_TARGET_GLES
+    if((data.flags & FlatGL2D::Flag::ObjectId) && !GL::Context::current().isExtensionSupported<GL::Extensions::EXT::gpu_shader4>())
+        CORRADE_SKIP(GL::Extensions::EXT::gpu_shader4::string() << "is not supported.");
+    if((data.flags & FlatGL2D::Flag::TextureArrays) && !GL::Context::current().isExtensionSupported<GL::Extensions::EXT::texture_array>())
+        CORRADE_SKIP(GL::Extensions::EXT::texture_array::string() << "is not supported.");
+    #endif
+
+    auto compileState = FlatGL<dimensions>::compile(data.flags);
+    CORRADE_COMPARE(compileState.flags(), data.flags);
+
```

Comment:
I'd like to have the actual async compilation tested here -- to make sure there was no brainfart involved that makes it never finish, and to have this use case represented with some actually working code people can look at.

```cpp
while(!compileState.isLinkFinished())
    Utility::System::sleep(100);
    
FlatGL<dimensions> shader{std::move(compileState)};
```

That would also make the `CORRADE_VERIFY(shader.isLinkFinished());` unneeded.

Same for the uniform buffer variant below.